protocol/contracts/beanstalk/sun/SeasonFacet/Receiving.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {C} from "contracts/C.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {AppStorage, Storage} from "contracts/beanstalk/AppStorage.sol";
+import {LibFertilizer} from "contracts/libraries/LibFertilizer.sol";
+import {ReentrancyGuard} from "contracts/beanstalk/ReentrancyGuard.sol";
+
+/**
+ * @title Receiving
+ * @author funderbrker
+ * @notice Holds the functions responsible for receiving Bean shipments after mints. These
+ *  functions must live inside of the Beanstalk Diamond. If new receiving components are needed,
+ *  this contract and its containing Facet will need to be updated.
+ * @dev An alternative design could remove the need for the generalized receive() entry function
+ *  and instead require the shipping route to define the selector of its own corresponding receive
+ *  function. However, both designs will require a Facet cut if a new receive function is needed,
+ *  so this design was chosen for additional clarity.
+ */
+contract Receiving is ReentrancyGuard {
+    using SafeCast for uint256;
+
+    /**
+     * @notice General entry point to receive Beans at a given component of the system.
+     * @param recipient The Beanstalk component that will receive the Beans.
+     * @param amount The amount of Beans to receive.
+     * @param data Additional data to pass to the receiving function.
+     */
+    function receiveShipment(
+        Storage.ShipmentRecipient recipient,
+        uint256 amount,
+        bytes memory data
+    ) internal {
+        if (recipient == Storage.ShipmentRecipient.Silo) {
+            siloReceive(amount, data);
+        } else if (recipient == Storage.ShipmentRecipient.Field) {
+            fieldReceive(amount, data);
+        } else if (recipient == Storage.ShipmentRecipient.Barn) {
+            barnReceive(amount, data);
+        }
```

Comment:
can we add a comment on the bottom indiciating that new ShipmentRecipient enums should be added here?

Reply:
https://github.com/BeanstalkFarms/Beanstalk/pull/860/commits/d65b6bacc7e6d96105a48c759af2f5222b0ccb26#diff-6c2813749618ebaa2ded478935db44e4d09054278e07e6ed5a595a1a6f7ab178R27